### DISCUSSION TITLE
What is the best practice for modifying an crate?

### DISCUSSION THREAD

**Discussion by Joebeazelman** [alire-project/alire#1424] [Q&A] [answered]
My project and its crates depends on another official Alire crate. Its source will be modified throughout the development process and regularly submitted as pull requests. I prefer to keep this crate as a separate project without breaking any dependencies while also retaining Alire's dependency and project management facilities.  What's the best practices for this type of workflow?

**Comment by mgrojo** [comment, marked as answer]
I'm not sure if I understand your needs, but maybe pins is the answer to your question: https://alire.ada.dev/docs/#work-in-progress-dependency-overrides

If that's not the case, examples or references to public repositories would help to understand your case.
